### PR TITLE
fix bug that cause dmhy scrapper leak memory

### PR DESCRIPTION
**Pull request by EverettSummer**
After investigation. I found that the scrapper doesn't close the Page object properly. cause an memory issue.

beside that:
fix potential issue which may cause scrapper exception stop the whole application.

### CODE REVIEW

File: src/scraper/dmhy.ts [hunk 1]
```diff
@@ -60,128 +62,149 @@ export class DmhyScraper implements Scraper {
...
+                mediaFile.name = basename(mediaFile.path, mediaFile.ext);
+                item.files.push(mediaFile);
             }
-            mediaFile.ext = extname(mediaFile.path);
-            mediaFile.name = basename(mediaFile.path, mediaFile.ext);
-            item.files.push(mediaFile);
+            return item;
+        } catch (e) {
+            console.error(e);
+            return null;
```
Comment (acgotaku) [inline src/scraper/dmhy.ts]: why add return?

File: src/scraper/bangumi-moe.ts [hunk 2]
```diff
@@ -26,14 +26,25 @@ export class BangumiMoe implements Scraper {
 
     public async executeTask(task: Task): Promise<any> {
         if (task.type === TaskType.SUB) {
-            return this._getTorrentDetail((task as BangumiMoeTask).item);
+            let item = await this._getTorrentDetail((task as BangumiMoeTask).item);
```
[the remark this answers is not in the document]
Comment (EverettSummer) [reply src/scraper/bangumi-moe.ts]: const almost has nothing to do with CONSTANT inside a function. A variable is declared with const doesn’t mean it’s immutable, all it means is the value can’t be re-assigned.

let's stop let vs const debate.
in this case use let or const doesn't matter

File: src/scraper/dmhy.ts [hunk 3]
```diff
@@ -60,128 +62,149 @@ export class DmhyScraper implements Scraper {
                 this._taskOrchestra.queue(newTask);
             }
         } else {
-            return this.scrapDetailPage(this._browser, (task as DmhyTask).item);
+            let item = await this.scrapDetailPage(this._browser, (task as DmhyTask).item);
```
[the remark this answers is not in the document]
Comment (EverettSummer) [reply src/scraper/dmhy.ts]: A variable is declared with const doesn’t mean it’s immutable, all it means is the value can’t be re-assigned. 
Only useful case is in top level declare. in method it's not neccessary use it everywhere. 

Let's stop the let vs const debate

File: src/scraper/dmhy.ts (hunk 1 above)
Comment (EverettSummer) [reply src/scraper/dmhy.ts]: The function should return something when a exception is raised. explicity return has better readiblity.

**Pull request merged**